mlflow/doctor.py

```diff
@@ -0,0 +1,47 @@
+import os
+import json
+import pkg_resources
+import platform
+import click
+import mlflow
+
+
+def doctor():
```

Comment:
The name `doctor` comes from `homebrew doctor` Any name that's short and easy to remember should be OK.

Reply:
scikit-learn provides a similar API: https://scikit-learn.org/stable/modules/generated/sklearn.show_versions.html

Reply:
The name sounds good to me :)